### ISSUE TITLE
[ REFACTOR ] /run/run-jobs-in-a-session

### ISSUE THREAD

**Issue by javabster** [Qiskit/documentation#258]

**Comment by jyu00** [comment]
>  clarify when sessions get automatically closed (first example doesn't explain enough)

Yeah we forgot to add `session.close()` in the code example. It now points out you need to manually close it if not using a context manager:

<img width="845" alt="image" src="https://github.com/Qiskit/documentation/assets/28284230/709fdb48-dbe8-4e6b-bb82-9b5de6700548">


We didn't add `session.close()` in the first code example because that's under "Open a session", but we certainly could.

>  clarify difference between close and cancel and when to use each. 

You close a session when you're done submitting jobs to it, so that the scheduler doesn't wait for interactive TTL to timeout. In this case pending jobs would still run to completion.  
You cancel a session if you want to quickly fail all pending jobs.
